metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -305,13 +305,8 @@ class DebugProvider(
           symbol = symbolInfo.symbol
           testClass <- testClasses.get(symbol)
         } yield testClass
-        val mains = for {
-          symbolInfo <- textDocument.symbols
-          symbol = symbolInfo.symbol
-          mainClass <- classes.get(symbol)
-        } yield mainClass
-        if (mains.nonEmpty) {
-          verifyMain(buildTarget, mains.toList, params)
+        if (classes.nonEmpty) {
+          verifyMain(buildTarget, classes.values.toList, params)
```

Comment:
Doesn't this mean we will get all the main classes in the project? I think this is why I added the extra step, but it seems that the symbol is maybe wrongly added to `classes` ?

Reply:
Ahhh duh, of course. I totally blanked on that. Yes, this does mean you get everything for the entire target since that's how `run` works.  🤔 I'll keep digging to see what's going on then.

Reply:
Ahhh the symbol is `_empty_/Main$package.hello().` whereas the main classes have `_empty_/hello.